finagle-redis/src/main/scala/com/twitter/finagle/Redis.scala

```diff
@@ -63,4 +66,48 @@ object Redis extends Client[Command, Reply] {
 
   def newService(dest: Name, label: String): Service[Command, Reply] =
     client.newService(dest, label)
+
+  object Subscribe extends com.twitter.finagle.Client[SubscribeCommand, Unit] {
+
+    object SubscribeClient {
+      /**
+       * Default stack parameters used for redis client.
+       */
+      val defaultParams: Stack.Params = StackClient.defaultParams +
+        param.ProtocolLibrary("redis.subscribe")
+
+      /**
+       * A default client stack which supports the pipelined redis client.
+       */
+      def newStack: Stack[ServiceFactory[SubscribeCommand, Unit]] = StackClient.newStack
+        .replace(DefaultPool.Role, SingletonPool.module[SubscribeCommand, Unit])
+    }
+
+    case class SubscribeClient(
```

Comment:
hmm, interesting.  I guess it depends on the implementation–the downside is that if you send the parts of a transaction eagerly instead of lazily, then if some of the work is nontrivial, like sending a big request, you don't delay it until the last moment.  especially since finagle typically moves IO to a different thread, this would be a bummer.